sklearn/neural_network/tests/test_mlp.py

```diff
@@ -718,3 +718,65 @@ def test_early_stopping_stratified():
             ValueError,
             match='The least populated class in y has only 1 member'):
         mlp.fit(X, y)
+
+
+def test_mlp_classifier_dtypes_casting():
+    mlp_64 = MLPClassifier(alpha=1e-5,
+                           hidden_layer_sizes=(5, 3),
+                           random_state=1, max_iter=100)
+    mlp_64.fit(X_digits[:300], y_digits[:300])
+    pred_64 = mlp_64.predict(X_digits[300:])
+
+    mlp_32 = MLPClassifier(alpha=1e-5,
+                           hidden_layer_sizes=(5, 3),
+                           random_state=1, max_iter=100)
+    mlp_32.fit(X_digits[:300].astype(np.float32), y_digits[:300])
+    pred_32 = mlp_32.predict(X_digits[300:].astype(np.float32))
+
+    assert_array_equal(pred_64, pred_32)
+
+
+def test_mlp_regressor_dtypes_casting():
+    mlp_64 = MLPRegressor(alpha=1e-5,
+                          hidden_layer_sizes=(5, 3),
+                          random_state=1, max_iter=50)
+    mlp_64.fit(X_digits[:300], y_digits[:300])
+    pred_64 = mlp_64.predict(X_digits[300:])
+
+    mlp_32 = MLPRegressor(alpha=1e-5,
+                          hidden_layer_sizes=(5, 3),
+                          random_state=1, max_iter=50)
+    mlp_32.fit(X_digits[:300].astype(np.float32), y_digits[:300])
+    pred_32 = mlp_32.predict(X_digits[300:].astype(np.float32))
+
+    assert_allclose(pred_64, pred_32, rtol=1e-04)
+
+
+def test_mlp_param_dtypes():
```

Comment:
This can be parametrize:

```py
@pytest.mark.parametrize('dtype', [np.float32, np.float64])
def test_mlp_param_dtypes(dtype):
	# Checks input dtype is used for attributes and prediction

	X, y = X_digits[:300].astype(dtype), y_digits[:300]
    mlp = MLPRegressor(alpha=1e-5,
                          hidden_layer_sizes=(5, 3),
                          random_state=1, max_iter=50)
    mlp.fit(X, y)
    pred = mlp_64.predict(X)

	assert pred.dtype == dtype
	...

Reply:
Done!